tests/test-books-api.js

```diff
@@ -0,0 +1,178 @@
+            let response = await axios.get(`${process.env.API_URL}/books/1`).catch(error => {
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+                assert.strictEqual(error.response.status, 500)
+            })
+        })
+
+        it('4. should return status 200 and the data response will give a null array when called with the wrong id\'s value', async() => {
+            let new_id = null;
+            if(book_id.charAt(35) == '1') {
+                new_id = book_id.substring(0,35) + '2'
+            } else {
+                new_id = book_id.substring(0,35) + '1'
+            }
+            let response = await axios.get(`${process.env.API_URL}/books/${new_id}`)
+            assert.strictEqual(response.status, 200)
+            assert.strictEqual(response.data.length, 0)
+        })
+    })
+
+    describe('create a book', async() => {
+        it('5. should return status 201 and the data response will give a correct created id when called', async() => {
+            let response = await axios.post(`${process.env.API_URL}/books`, {
+                isbn: isbn,
+                name: name,
+                author: author,
+                publisher: publisher,
+                quantity: quantity,
+                id_category: id_category
+            })
+            assert.strictEqual(response.status, 201)
+            assert.strictEqual(response.data.State, State.SUCCESS)
+
+            let result = await dbbooks.get_book_by_id_async(response.data.createdBookId)
+            assert.strictEqual(result.rows[0].isbn, isbn)
+            assert.strictEqual(result.rows[0].name, name)
+            assert.strictEqual(result.rows[0].author, author)
+            assert.strictEqual(result.rows[0].publisher, publisher)
+            assert.strictEqual(result.rows[0].quantity, quantity)
+            assert.strictEqual(result.rows[0].id_category, id_category)
+        })
+
+        it('6.should return status 500 and the State response will be error when called with no parameters', async() => {
+            let response = await axios.post(`${process.env.API_URL}/books`, {}).catch(error => {
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+        })
+    })
+
+    describe('update Book', async() => {
+        it('7.should return status 200 and the data in database is the same with the data response when called', async() => {
+            const book_isbn = 'jlasdjf9381234'
+            const book_name = 'Naruto'
+            const book_author = 'Masashi Kishimoto'
+            const book_publisher = 'Kim Dong'
+            const book_quantity = 230
+            const book_id_category = '1c2ad358-98b0-4210-9ea3-3c4625b76e91'
```

Comment:
Should not use a specific Id in testing because we can't assume the database has already haven it. Add new entity if you want to use it.

Reply:
I fixed it